crates/host/src/wasmbus/mod.rs

```diff
@@ -3493,103 +3553,8 @@ impl Host {
-                    return None;
-                }
-                let name = actor
-                    .claims()
-                    .and_then(|claims| claims.metadata.as_ref())
-                    .and_then(|metadata| metadata.name.as_ref())
-                    .cloned();
-                Some(ActorDescription {
-                    id: id.into(),
-                    // SAFETY: We just checked above and returned `None` if instances were empty
-                    // Preserve backwards compatible payload by returning the first image reference
-                    image_ref: instances
-                        .first()
-                        .expect("actor should have at least one instance")
-                        .image_ref
-                        .clone(),
-                    instances,
-                    name,
-                })
-            })
-            .collect()
-            .await;
-        let providers = self.providers.read().await;
-        let providers: Vec<_> = providers
-            .iter()
-            .filter_map(
-                |(
-                    id,
-                    Provider {
-                        claims,
-                        instances,
-                        image_ref,
-                        ..
-                    },
-                )| {
-                    let jwt::CapabilityProvider {
-                        capid: contract_id,
-                        name,
-                        rev: revision,
-                        ..
-                    } = claims.metadata.as_ref()?;
-                    Some(instances.iter().map(
-                        move |(link_name, ProviderInstance { annotations, .. })| {
-                            let annotations = Some(annotations.clone().into_iter().collect());
-                            let revision = revision.unwrap_or_default();
-                            ProviderDescription {
-                                id: id.into(),
-                                image_ref: Some(image_ref.clone()),
-                                contract_id: contract_id.clone(),
-                                link_name: link_name.into(),
-                                name: name.clone(),
-                                annotations,
-                                revision,
-                            }
-                        },
-                    ))
-                },
-            )
-            .flatten()
-            .collect();
-        let buf = serde_json::to_vec(&HostInventory {
-            host_id: self.host_key.public_key(),
-            issuer: self.cluster_key.public_key(),
-            labels: self.labels.read().await.clone(),
-            friendly_name: self.friendly_name.clone(),
-            actors,
-            providers,
-        })
-        .context("failed to encode reply")?;
+        let inventory = self.inventory().await;
+        let buf = serde_json::to_vec(&inventory).context("failed to encode reply")?;
```

Comment:
```suggestion
        let buf = serde_json::to_vec(&inventory).context("failed to encode inventory")?;
```

Better to be more specific?